### PR TITLE
cmd/{k8s-operator,containerboot},kube/egressservices: fix Pod IP check for dual stack clusters

### PR DESCRIPTION
**Pull request by irbekrm**
Currently egress Services for ProxyGroup only work for Pods and Services with IPv4 addresses. Ensure that it works on dual stack clusters by reading proxy Pod's IP from the `.status.podIPs` list that always contains both IPv4 and IPv6 address (if the Pod has them) rather than `.status.podIP` that, in a dual stack cluster, could contain the IPv6 address of the Pod.

Updates tailscale/tailscale#13406

### CODE REVIEW

File: cmd/containerboot/settings.go [hunk 1]
```diff
@@ -62,11 +63,67 @@ type settings struct {
...
+		Root:                                  defaultEnv("TS_TEST_ONLY_ROOT", "/"),
+		TailscaledConfigFilePath:              tailscaledConfigFilePath(),
+		AllowProxyingClusterTrafficViaIngress: defaultBool("EXPERIMENTAL_ALLOW_PROXYING_CLUSTER_TRAFFIC_VIA_INGRESS", false),
+		PodIP:                                 defaultEnv("POD_IP", ""),
+		EnableForwardingOptimizations:         defaultBool("TS_EXPERIMENTAL_ENABLE_FORWARDING_OPTIMIZATIONS", false),
+		HealthCheckAddrPort:                   defaultEnv("TS_HEALTHCHECK_ADDR_PORT", ""),
+		EgressSvcsCfgPath:                     defaultEnv("TS_EGRESS_SERVICES_CONFIG_PATH", ""),
+	}
+	podIPs, ok := os.LookupEnv("POD_IPS")
+	if ok {
```
Comment (tomhjp) [inline cmd/containerboot/settings.go:104]: Does it make sense to fall back to `POD_IP` if `POD_IPS` is not set? And similarly, have `PodIP` fall back to the first IP address from `PodIPs` if not set? I don't have a concrete use-case in mind right now, but they seem like reasonable backup values and would represent the status yaml more closely.

File: cmd/k8s-operator/egress-eps.go [hunk 2]
```diff
@@ -142,14 +156,21 @@ func (er *egressEpsReconciler) podIsReadyToRouteTraffic(ctx context.Context, pod
 		l.Debugf("proxy Pod is being deleted, ignore")
 		return false, nil
 	}
-	podIP := pod.Status.PodIP
+	podIP, err := podIPv4(&pod)
+	if err != nil {
+		return false, fmt.Errorf("error determining Pod IP address: %v", err)
+	}
+	if podIP == "" {
+		l.Infof("[unexpected] Pod does not have an IPv4 address, this will most likely not work")
```
Comment (tomhjp) [inline cmd/k8s-operator/egress-eps.go:164]: Given that we're returning early, I think we can be stronger than "most likely" :)

```suggestion
		l.Infof("[unexpected] Pod does not have an IPv4 address, and IPv6 is not currently supported")
```

**Review by tomhjp — APPROVED** [review]
LGTM!

File: cmd/containerboot/settings.go (hunk 1 above)
Comment (irbekrm) [reply cmd/containerboot/settings.go:104]: My assumption is that the `.status.podIP` predates `.status.podIPs` and is there for backwards compatibility reasons. I can take a look at the upstream code to see how it's set. From a brief test it seems like the list is always set.
If we do discover an edge case where `.status.podIPs` is not set, but `.status.podIP` is we can always implement a fallback without breaking anyone for whom it's working

Comment (tomhjp) [reply cmd/containerboot/settings.go:104]: Yeah I think that's accurate. As I say, I don't have a strong case for doing this, just felt like an easy "may as well", but I am also totally fine with not doing it

Comment (irbekrm) [reply cmd/containerboot/settings.go:104]: Added a TODO to double check this 👍🏼
